sklearn/metrics/pairwise.py

```diff
@@ -256,6 +284,123 @@ def euclidean_distances(X, Y=None, Y_norm_squared=None, squared=False,
     return distances if squared else np.sqrt(distances, out=distances)
 
 
+def masked_euclidean_distances(X, Y=None, squared=False,
+                               missing_values="NaN", copy=True):
+    """Calculates euclidean distances in the presence of missing values
+
+    Considering the rows of X (and Y=X) as samples, compute the distance matrix
```

Comment:
I find the docstring unnecessarily complicated. ``euclidean_distances`` is simpler, right? How about ``Computes the euclidean distance between each pair of rows in X and Y. If Y=None, Y=X is assumed".